nixos/tmate-ssh-server: init module

jlesquembre opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

jlesquembre committed:
nixos/tmate-ssh-server: init module

ck3d reviewed (approved):
LGTM

aanderse commented on nixos/modules/services/networking/tmate-ssh-server.nix:
```diff
@@ -0,0 +1,110 @@
+  };
+
+  config = mkIf cfg.enable {
+    networking.firewall.allowedTCPPorts = [ cfg.port ];
```
Please guard this being an `openFirewall` option.

jlesquembre replied:
Added in https://github.com/NixOS/nixpkgs/pull/192270/commits/0844d23e6f507ecea507f04f9f5f383bd62f1f56

mkaito commented:
@ofborg test tmate-ssh-server
@ofborg build tmate-ssh-server

mkaito commented:
Likely want to add yourself to maintainers.

The tests fail, but ofborg seems to consider that a "neutral outcome". Wanna try and fix them?

jlesquembre committed:
nixos/tmate-ssh-server: add openFirewall option

jlesquembre committed:
nixos/tmate-ssh-server: fix tests

For some reason running the tests with
nix run .#nixosTests.tmate-ssh-server.driver
worked, but running with
nix-build ./nixos/release.nix -A tests.tmate-ssh-server
did not.

jlesquembre commented:
@ofborg test tmate-ssh-server
@ofborg build tmate-ssh-server

jlesquembre commented:
For some reason running the tests with `nix run .#nixosTests.tmate-ssh-server.driver`
worked, but running with `nix-build ./nixos/release.nix -A tests.tmate-ssh-server` did not. With the changes in https://github.com/NixOS/nixpkgs/pull/192270/commits/0627875e671c0e2660ccb9236513ec0209b6c0a6 should work on both cases

mkaito reviewed (approved):
lgtm

aanderse commented on nixos/modules/services/networking/tmate-ssh-server.nix:
```diff
@@ -0,0 +1,122 @@
+
+  config = mkIf cfg.enable {
+
+    networking.firewall.allowedTCPPorts = if cfg.openFirewall then [ cfg.port ] else [ ];
```
Entirely fine as is, but can also be written as:

```suggestion
    networking.firewall.allowedTCPPorts = optionals cfg.openFirewall [ cfg.port ];
```

jlesquembre replied:
Thanks for the suggestion, I find it more readable.

aanderse reviewed (approved):
Module LGTM :+1:

jlesquembre committed:
nixos/tmate-ssh-server: use optionals

Co-authored-by: Aaron Andersen <aaron@fosslib.net>
